src/components/MobileNav.js

```diff
@@ -131,6 +170,18 @@ const MobileNav = ({ networkId, viewedAddress }) => {
     return clearAllBodyScrollLocks;
   }, [sidebarDrawerOpen]);
 
+  useEffect(() => {
```

Comment:
This is not a correct use of `useEffect`. Note that `useEffect` [runs *after* rendering](https://reactjs.org/docs/hooks-reference.html#useeffect). Here you can just do this work directly in the body of the function. The key difference from other code that uses `useEffect` is that we are not making any async calls here.

It then follows that there is also no need for `useState`, because you can just directly construct your icon data object and pass it into the child component.

i.e. (omitting unrelated details):
```js
const MobileNav = ({ cdpId, ...otherProps }) => {
  let iconData;

  if (cdpId) {
    iconData = foo;
  } else {
    iconData = bar;
  }

  return <ChildComponent iconData={iconData}/>;
}
```

Reply:
Removed `useEffect` here, and it it works nicely.